policies/coding-style.md

```diff
@@ -584,28 +594,93 @@ less complicated than coding pure assembly, and it doesn't provide the
 same performance guarantee across different micro-architecture. Nor is
 it portable enough to meet our multi-platform support goals.
 
-## Chapter 14:  Portability
+## Chapter 14: Portability
 
 To maximise portability the version of C defined in ISO/IEC 9899:1990
 should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
-## Chapter 15: Miscellaneous
+## Chapter 15: Expressions
 
-Do not use `!` to check if a pointer is NULL, or to see if a str...cmp
-function found a match.  For example, these are wrong:
+Avoid needless parentheses as far as reasonable.
+For example, do not write
+```c
+    if ((p == NULL) && (!f(((2 * x) + y) == (z++))))
+```
+but
+```c
+    if (p == NULL && !f(2 * x + y == z++)).
+```
 
+For clarity, always put parentheses when mixing `&&` and `||` operations,
+comparison operators like `<=` and `==`, and bitwise operators.
+For example,
 ```c
-    if (!(p = BN_new())) ...
-    if (!strcmp(a, "FOO")) ...
+    if ((a && b) || c)
+    if ((a <= b) == ((c >= d) != (e < f)))
+    x = (a & b) ^ (c | d)
 ```
 
-Do this instead:
+Regarding parentheses in macro definitions see the
+[chapter on macros](#chapter-9-macros-and-enums).
+
+In comparisons with constants (including `NULL` and other constant macros)
+place the constant on the right-hand side of the comparison operator.
+For example,
+```c
+    while (i++ < 10 && p != NULL)
+```
 
+Do not use implicit checks for
+numbers (not) being `0` or pointers (not) being `NULL`.
```

Comment:
Where the number is intended to be a boolean value I think this is ok and is *very* common in the codebase (in particular when checking function return values)m e.g. `if (!EVP_DecryptUpdate(...))`

Reply:
Sure, and therefore the rule is stated just about *numbers* (integers etc.), not about Boolean values.
Maybe this should be made more clear.

Reply:
@mattcaswell it would have been good to receive such comments regarding clarifications earlier,
not after most people have already voted.

Reply:
To clarify this, we could add (after the given examples) for instance:

Note that Boolean values shall be used directly as usual, e.g.,
`if (check(x) && !success(y))`